cmd/nitro/nitro.go

```
@@ -267,6 +270,16 @@ func mainImpl() int {
 			flag.Usage()
 			log.Crit("error generating HMAC key for Espresso Caff node", "err", err)
 		}
+		privHex := hex.EncodeToString(key.D.Bytes())
+		// This will be used by the hyperlane validator
+		os.Setenv("SNAPSHOT_PRIVATE_KEY", privHex)
+		if nodeConfig.ParentChain.ID != 0 {
```

I dont understand why we need this check to see if parent chain id is zero or not

---

I think this was just to ensure a valid tx opts as there isn't a chainid with value 0, which would also be the value if the chainID isn't set in the config.

---

I think if nodeConfig.ParentChain.ID == 0 we should throw an error right?